fsuae-launcher: ModuleNotFoundError: No module named 'distutils'

### Describe the bug
fsuae-launcher installation fails with the following error:

```
...
python3 setup.py install         --install-lib=/nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/fs-uae-launcher  --install-scripts=/nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/fs-uae-launcher
Traceback (most recent call last):
  File "/build/fs-uae-launcher-3.1.70/setup.py", line 6, in <module>
    from distutils.core import setup
ModuleNotFoundError: No module named 'distutils'
make: *** [Makefile:107: install-program] Error 1
error: builder for '/nix/store/m2qh45skcly1szrnmxaqh4id3fjzkb0m-fs-uae-launcher-3.1.70.drv' failed with exit code 2;
       last 25 log lines:
       > mkdir -p share/locale/ru/LC_MESSAGES
       > msgfmt --verbose po/ru.po -o share/locale/ru/LC_MESSAGES/fs-uae-launcher.mo
       > 412 translated messages, 71 untranslated messages.
       > mkdir -p share/locale/sr/LC_MESSAGES
       > msgfmt --verbose po/sr.po -o share/locale/sr/LC_MESSAGES/fs-uae-launcher.mo
       > 254 translated messages, 229 untranslated messages.
       > mkdir -p share/locale/sv/LC_MESSAGES
       > msgfmt --verbose po/sv.po -o share/locale/sv/LC_MESSAGES/fs-uae-launcher.mo
       > 0 translated messages, 483 untranslated messages.
       > mkdir -p share/locale/tr/LC_MESSAGES
       > msgfmt --verbose po/tr.po -o share/locale/tr/LC_MESSAGES/fs-uae-launcher.mo
       > 473 translated messages, 10 untranslated messages.
       > Running phase: glibPreInstallPhase
       > Running phase: installPhase
       > install flags: SHELL=/nix/store/717iy55ncqs0wmhdkwc5fg2vci5wbmq8-bash-5.2p32/bin/bash prefix=\$\(out\) gsettingsschemadir=/nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/gsettings-schemas/fs-uae-launcher-3.1.70/glib-2.0/schemas/ install
       > mkdir -p /nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share
       > cp -a share/* /nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share
       > mkdir -p /nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/doc/fs-uae-launcher
       > cp -a README COPYING /nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/doc/fs-uae-launcher
       > python3 setup.py install        --install-lib=/nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/fs-uae-launcher  --install-scripts=/nix/store/yzqxp4ps19x2i2g7mr3x53l28fqc4l1s-fs-uae-launcher-3.1.70/share/fs-uae-launcher
       > Traceback (most recent call last):
       >   File "/build/fs-uae-launcher-3.1.70/setup.py", line 6, in <module>
       >     from distutils.core import setup
       > ModuleNotFoundError: No module named 'distutils'
       > make: *** [Makefile:107: install-program] Error 1
       For full logs, run 'nix log /nix/store/m2qh45skcly1szrnmxaqh4id3fjzkb0m-fs-uae-launcher-3.1.70.drv'.
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix-shell -p fsuae-launcher`

### Notify maintainers

@svanderburg @AndersonTorres

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.57, NixOS, 24.11 (Vicuna), 24.11.20241018.4c2fcb0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/store/riqkpszjqk02bi1wppfg8ip5xvh102qd-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fs/fsuae-launcher/package.nix
+++ b/pkgs/by-name/fs/fsuae-launcher/package.nix
@@ -34,6 +34,11 @@ stdenv.mkDerivation (finalAttrs: {
 
   dontWrapQtApps = true;
 
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail "distutils.core" "setuptools"
+  '';
+
   preFixup = ''
     wrapQtApp "$out/bin/fs-uae-launcher" \
       --set PYTHONPATH "$PYTHONPATH"

